### ISSUE TITLE
[Suggestion] Use system launcher instead of Anbox Application Manager

### ISSUE THREAD

**Issue by Skqnder** [anbox/anbox#749]
Wouldn't be _more native_ to use the desktop environment's application launcher instead of `Anbox Application Manager`?
I see that there are already `.desktop` files for each installed apk under `~/snap/anbox/common/app-data/applications/anbox`, so why not symlink those to `~/.local/share/applications`? Then create a new Anbox category for category-based application menus and an Anbox [app folder](https://wiki.gnome.org/HowDoI/AppFolders) for GNOME's overview.

**Comment by morphis** [comment]
You're right, we need a better integration with the outer system but sadly snaps don't allow (yet) to provide dynamically created .desktop files to the outer system. Because of this we can't do the symlink automatically from within the snap.

That said, there are multiple things which are possible: 

 * Package Android applications as snaps. This is very attractive as people can then distribute Android applications via the snap store as they were regular apps. Each snap then provides exactly one .desktop file to the host system and integrates quite well. 
 * Discuss with snapd upstream if dynamic creation of .desktop files and distributing them to the host is acceptable and can be supported.
 * Another approach would be to provide a gnome shell search provider (see https://developer.gnome.org/SearchProvider/) from the Anbox snap. This way we can always populate a good list of applications to the shell. However even this is not possible without manually placing a file in `$(datadir)/gnome-shell/search-providers` but there is a thread discussion this (see https://forum.snapcraft.io/t/gnome-shell-search-providers-in-snaps/5265)

**Comment by Skqnder** [comment]
I'm not sure but I believe snaps' confinement prevents them from _talking_ to each other. Packaging Android applications as snaps will probably prevent them from reaching Anbox and packaging Anbox with each Android app will be overkill as each app will be over 400MB.
The GNOME shell search provider idea is cool but it won't allow for example pinning apps to the dock or creating separate icons on the desktop.
Another idea would be to create a separate "helper" application that gets installed from the PPA (along with the kernel modules) to just manage application icons dynamically.

**Comment by fhteagle** [comment]
As a quick and dirty I had set up a cron entry to symlink from .local/share/applications to  ~/snap/anbox/common/app-data/applications/anbox . This worked great in Gnome3 for about a day, but now the ~/snap/anbox/common/app-data/applications/anbox  directory is empty. Very odd, and probably an Anbox bug?!?

**Comment by GalGreenfield** [comment]
Hi, are there any news about it?

**Comment by steelcowboy** [comment]
> As a quick and dirty I had set up a cron entry to symlink from .local/share/applications to ~/snap/anbox/common/app-data/applications/anbox . This worked great in Gnome3 for about a day, but now the ~/snap/anbox/common/app-data/applications/anbox directory is empty. Very odd, and probably an Anbox bug?!?

Not sure why a symlink would cause problems, but a bind mount may be more effective. You can add the following to `/etc/fstab`, replacing <YOUR_USERNAME> with your username:
```
/home/<YOUR_USERNAME>/snap/anbox/common/app-data/applications/anbox
/home/<YOUR_USERNAME>/.local/share/applications/anbox none defaults,bind 0 0
```
Then do:
```
mkdir -p ~/.local/share/applications/anbox
sudo mount ~/.local/share/applications/anbox
```

**Comment by Fuseteam** [comment]
i'm quite interested in packaging android apps as snaps but it does make me wonder what would be needed for that
